### PR TITLE
Fix a minor issue with finding the files to run shellcheck against

### PR DESCRIPTION
**Pull request by baumanj**
The builder version of https://github.com/habitat-sh/habitat/pull/5396

Pretty much all the changes are just double-quotes to suppress unwanted expansion. There is one legit bug (occurring in several places) that I've called out with a comment.

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: .studiorc [hunk 1]
```diff
@@ -235,7 +235,7 @@ generate_bldr_keys() {
   keys=( $(find /hab/cache/keys -name "bldr-*.pub") )
 
   if [ "${#keys[@]}" -gt 0 ]; then
-    KEY_NAME=$(echo $keys[0] | grep -Po "bldr-\d+")
+    KEY_NAME=$(echo "${keys[0]}" | grep -Po "bldr-\d+")
```
Comment (baumanj) [inline .studiorc]: This syntax was just wrong. You always need braces to access an array variable:
```bash
$ arr=(one two three)
$ echo $arr[0]
one[0]
$ echo ${arr[0]}
one
```
The `-o` flag to `grep` just happened to save us.

File: components/builder-api/habitat/hooks/run [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
+# shellcheck disable=1083
+
 
 export HOME={{pkg.svc_data_path}}
 export RUST_LOG={{cfg.log_level}}
 export RUST_BACKTRACE=1
 pkg_svc_run="bldr-api start -c {{pkg.svc_config_path}}/config.toml"
 
-if [ "\$(whoami)" = "root" ]; then
+if [ "$(whoami)" = "root" ]; then
```
Comment (baumanj) [inline components/builder-api/habitat/hooks/run]: This is a bug. It also occurs several other places:
```
$ whoami
jbauman
$ if [ "$(whoami)" = "jbauman" ]; then echo "you're jbauman"; else echo "you're not jbuman"; fi
you're jbauman
$ if [ "\$(whoami)" = "jbauman" ]; then echo "you're jbauman"; else echo "you're not jbuman"; fi
you're not jbuman
```

**Comment by baumanj** [comment]
Looks like there are some issues here. I'm sorting them out.

**Pull request merged**